### PR TITLE
[WIP] Update client to be compatible for Python 3.12

### PR DESCRIPTION
**Pull request by petrpan26**
Adding python 3.12 as a version for fennel-ai client test CI/CD. Also try to update dependencies to compatible with python 3.12.

Upgrade a dependencies that fail in 3.12 and upgrade urllib vendoring.

### CODE REVIEW

File: fennel/_vendor/README.md
Comment (mohitreddy1996) [inline fennel/_vendor/README.md:1]: why remove this?

Comment (petrpan26) [reply fennel/_vendor/README.md:1]: I think when I run sync it remove this. Let me add this again hmm

File: fennel/_vendor/urllib3/util/ssl_.py [hunk 1]
```diff
@@ -375,18 +461,35 @@
...
+            # Import error, ssl is not available.
+            raise ProxySchemeUnsupported(
+                "TLS in TLS requires support for the 'ssl' module"
+            )
+
+        SSLTransport._validate_ssl_context_for_tls_in_tls(ssl_context)
+        return SSLTransport(sock, ssl_context, server_hostname)
+
+    if server_hostname:
+        return ssl_context.wrap_socket(sock, server_hostname=server_hostname)
```
[the remark this answers is not in the document]
Comment (petrpan26) [reply fennel/_vendor/urllib3/util/ssl_.py:493]: @mohitreddy1996 @aditya-nambiar should this be look into it seems like from vendor

**Comment by aditya-nambiar** [comment]
Can you mark 3.12 as a required test to pass

**Comment by aditya-nambiar** [comment]
Also upgrade the client version in pyproject.toml

**Comment by mohitreddy1996** [comment]
@petrpan26 can idea these alerts were? And were they automatically closed? 

image

**Review by aditya-nambiar — APPROVED** [review]

**Comment by petrpan26** [comment]
@mohitreddy1996 I ignored it since it was in the vendoring library. Not sure what is the best course here

**Comment by mohitreddy1996** [comment]
@petrpan26 can you check if such alerts did show up in urllib3 source code itself maybe?

**Comment by mohitreddy1996** [comment]
AFAIR, this is coming from - https://github.com/PyCQA/bandit?

**Comment by mohitreddy1996** [comment]
@petrpan26 looks like we do have such instances across our code base - 
image

Again mostly from vendored packages. 

This will still be good to do IMO - https://github.com/fennel-ai/client/pull/333#issuecomment-1885912537. Approving to unblock post some investigation :)

**Review by mohitreddy1996 — APPROVED** [review]

**Comment by petrpan26** [comment]
Hmm i saw the same code line in their codebase but they dont show up as security alert. There must be some way of reporting it.

**Pull request merged**